### PR TITLE
Added Support for different data types.

### PR DESCRIPTION
**Pull request by amitsrivastava78**
Added different data types along with test cases.

### CODE REVIEW

**Comment by amitsrivastava78** [comment]
@miaout17 , can you please review the PR.

Regards
Amit

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by gunan** [comment]
@rthadur, could you follow up on the review.

**Comment by rthadur** [comment]
@gunan sure , thanks for the reminder.

File: tensorflow/lite/kernels/if.cc [hunk 1]
```diff
@@ -129,16 +140,55 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
   const OpData* op_data = reinterpret_cast<OpData*>(node->user_data);
 
   const TfLiteTensor* cond = GetInput(context, node, 0);
-  bool cond_value = cond->data.b[0];
+
+  int active_branch_subgraph_index = -1;
+
+  switch (cond->type) {
```
Comment (miaout17) [inline tensorflow/lite/kernels/if.cc]: Is should be possible to significantly shorten the code, like...
```
bool condition;
switch (cond->type) {
    case kTfLiteInt32:
      condition = cond->data.i32[0];
      break;
    case kTfLiteFloat32:
      condition = cond->data.f[0];
      break;
    // omitted...
  }
  active_branch_subgraph_index = condition
                                         ? op_data->then_subgraph_index
                                         : op_data->else_subgraph_index;
  

```

**Review by miaout17 — CHANGES_REQUESTED** [review]
Thanks for the contribution. Mostly looks good. 

Side note: Usually we should bump the op version when adding supported types, but since this the conversion process is experimental and nobody seems to be using it, I think it's fine to stick with version 1.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 43 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

**Pull request closed without merging**